### PR TITLE
[android][tools] fix android versioning

### PR DESCRIPTION
**Pull request by Kudo**
# Why

fix android versioning after react-native 0.71 upgrade

# How

mostly fixing transforms

# Test Plan

1. `et add-sdk -p android -s 48.0.0`
2. `cd android ; ./gradlew :app:assembleVersionedDebug`
3. versioned android expo go + sdk 48 ncl smoke test

# Checklist

- n/a Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: tools/src/versioning/android/expoModulesTransforms.ts [hunk 1]
```diff
@@ -82,6 +82,16 @@ export function expoModulesTransforms(module: string, abiVersion: string): FileT
...
+    'expo-image': {
+      content: [
+        {
+          // glide only supports single AppGlideModule, we use the one from unversioned expo-image
+          paths: 'SourceMap.kt',
+          find: new RegExp(`\\bimport ${abiVersion}\\.expo\\.modules\\.image\\.GlideOptions`, 'g'),
+          replaceWith: 'import expo.modules.image.GlideOptions',
+        },
+      ],
+    },
```
Comment (lukmccall) [inline tools/src/versioning/android/expoModulesTransforms.ts:94]: Let's add an option not to include the AppGlideModule from the expo image and create a separate glide app for Expo Go. The `react-native-fast-image` for instance, has that option https://github.com/DylanVann/react-native-fast-image/blob/main/docs/app-glide-module.md. However, for now, we can leave it as it's.

**Review by lukmccall — APPROVED** [review]

File: android/settings.gradle [hunk 2]
```diff
@@ -39,6 +39,7 @@ project(":expo-random").projectDir = new File("../packages/expo-random/android")
 useExpoModules([
     exclude : [
         'expo-module-template',
+        'expo-in-app-purchases',
```
Comment (tsapeta) [inline android/settings.gradle:42]: So it was included in the previous version? 😮

**Review by tsapeta — APPROVED** [review]

File: tools/src/versioning/android/expoModulesTransforms.ts (hunk 1 above)
Comment (Kudo) [reply tools/src/versioning/android/expoModulesTransforms.ts:94]: we had this code but i cannot get it working.
https://github.com/expo/expo/blob/75c7ead1353d98508129058290b1271c6272fcf5/packages/expo-image/android/build.gradle#L60-L63

on the other hand, if we exclude AppGlideModule from expo-image, where should we put the AppGlideModule across sdks? e.g. `android/expoview/main/java/host/exp/exponent/` as shared AppGlideModule for all sdks?

File: android/settings.gradle (hunk 2 above)
Comment (Kudo) [reply android/settings.gradle:42]: i think no. there is a sanity check for versioning script: https://github.com/expo/expo/blob/75c7ead1353d98508129058290b1271c6272fcf5/tools/src/Packages.ts#L206-L223
and it previously matched to the standalone `useExpoModules` which [is removed in this commit](https://github.com/expo/expo/commit/21d457592eab92917d9091a27961a413a1177e84#diff-9a50ef936e92a26017fd739f42ee0a0e7f80aec39dafc0a13f27b2548ee5e51f).

File: tools/src/versioning/android/expoModulesTransforms.ts (hunk 1 above)
Comment (lukmccall) [reply tools/src/versioning/android/expoModulesTransforms.ts:94]: I'm not sure which place will be the best, but I would put it in the non-versioned code to share that across SDKs - probably in the kernel.

Comment (Kudo) [reply tools/src/versioning/android/expoModulesTransforms.ts]: #21015 doesn't actually work. kotlin compiler doesn't respect sourceSets excludes 🤷‍♂️
https://github.com/expo/expo/pull/20889/commits/9b2bdc8790863238dda44e31de8cd47a84740c51 is the alternative fix.

**Pull request merged**